### ISSUE TITLE
`class_ids` or `ignore_class`!

### ISSUE THREAD

**Issue by innat** [keras-team/keras-cv#1027]
They're same in nature, aren't they? We should follow one way.

https://www.tensorflow.org/api_docs/python/tf/keras/metrics/MeanIoU
https://www.tensorflow.org/api_docs/python/tf/keras/losses/SparseCategoricalCrossentropy
> **ignore_class:** `Optional integer. The ID of a class to be ignored during loss computation. This is useful, for example, in segmentation problems featuring a "void" class (commonly -1 or 255) in segmentation maps. By default (ignore_class=None), all classes are considered.

https://keras.io/api/keras_cv/metrics/coco_mean_average_precision/
> **class_ids:** The class IDs to evaluate the metric for. To evaluate for all classes in over a set of sequentially labelled classes, pass range(classes).

**Comment by LukeWood** [comment]
> They're same in nature, aren't they? We should follow one way.

I think the distinction is in the use cases.  In the metric, the goal is to compute MaP for a subset of classes.  
in the losses you typically want a way to tell the model to ignore a specific subset of the tensor.

IMO we don't need to standardize on these, feel free to re-open if you disagree

**Comment by innat** [comment]
@LukeWood 
Do you generally mean that, for metric (ignore_class) and for loss (class_ids) ?

**Comment by LukeWood** [comment]
> @LukeWood Do you generally mean that, for metric (ignore_class) and for loss (class_ids) ?

I think for metrics it is probably case by case - but I'd expect class_ids to be more common.

For losses I do think ignore_class is almost always sufficient; you can update your model to use a `tf.where()` statement to mask out classes you want to ignore if you're using the loss.
